### PR TITLE
docs(textfield): automate documentation for markdown api tables

### LINKED ISSUE #4905: Add auto-docs to textfield and top-app-bar
- get text field and top app bar working on the new auto documentation generation tool 
- move all docs from README.md inline to TS files.

### PR DESCRIPTION
**Pull request by moog16**
fixes #4905

### CODE REVIEW

File: packages/mdc-textfield/README.md [hunk 1]
```diff
@@ -329,102 +329,155 @@ Mixin | Description
...
-* `minLength`
-* `maxLength`
-* `min`
-* `max`
-* `step`
-
-Method Signature | Description
+Signature | Description
 --- | ---
+`emit(evtType: string, evtData: T, shouldBubble?: boolean) => void` | Fires a cross-browser-compatible custom event from the component root of the given type, with the given data.
```
Comment (abhiomkar) [inline packages/mdc-textfield/README.md]: We might've to ignore emit, listen and unlisten which are not useful to clients.

File: packages/mdc-textfield/README.md [hunk 2]
```diff
@@ -329,102 +329,155 @@ Mixin | Description
...
+`listen(evtType: K, handler: SpecificEventListener<K>, options?: AddEventListenerOptions | boolean) => void` | Wrapper method to add an event listener to the component's root element. This is most useful when listening for custom events.
+`unlisten(evtType: K, handler: SpecificEventListener<K>, options?: AddEventListenerOptions | boolean) => void` | Wrapper method to remove an event listener to the component's root element. This is most useful when unlistening for custom events.
 
-## Usage Within Frameworks
+#### Properties
 
-If you are using a JavaScript framework, such as React or Angular, you can create a Text Field for your framework. Depending on your needs, you can use the _Simple Approach: Wrapping MDC Web Vanilla Components_, or the _Advanced Approach: Using Foundations and Adapters_. Please follow the instructions [here](../../docs/integrating-into-frameworks.md).
+Name | Type | Description
+--- | --- | ---
+pattern | `string` | @param Sets the input element's validation pattern.
```
Comment (abhiomkar) [inline packages/mdc-textfield/README.md]: Back tick is missing.

File: packages/mdc-textfield/README.md [hunk 3]
```diff
@@ -329,102 +329,155 @@ Mixin | Description
...
+`setTransformOrigin(evt: TouchEvent | MouseEvent) => void` | Sets the line ripple's transform origin, so that the line ripple activate animation will animate out from the user's click location.
+`setUseNativeValidation(useNativeValidation: boolean) => void` | Sets whether to check native HTML validity state (`true`, default) or custom validity state when updating styles (`false`). Enables or disables the use of native validation.
+`setValid(isValid: boolean) => void` | Sets custom validity and updates styles accordingly. Note that native validation will still be honored subsequently unless `setUseNativeValidation(false)` is also called.
+`setValue(value: string) => void` | Sets the input's value.
+
+#### Properties
+
+Name | Type | Description
+--- | --- | ---
+shouldFloat | `boolean` | Determines whether the label should float.
```
Comment (abhiomkar) [inline packages/mdc-textfield/README.md]: Missing backticks.

File: packages/mdc-textfield/README.md (hunk 1 above)
Comment (moog16) [reply packages/mdc-textfield/README.md]: Technically a user can use the emit, listen, and unlisten methods. I think leaving this out would make the list incomplete.

File: packages/mdc-textfield/README.md (hunk 2 above)
Comment (moog16) [reply packages/mdc-textfield/README.md]: Are you saying there should be backticks around the method or property name?

File: packages/mdc-textfield/README.md (hunk 3 above)
Comment (abhiomkar) [reply packages/mdc-textfield/README.md]: Missing backticks for `shouldFloat` (property name).

**Comment by moog16** [comment]
updated! ready for rereview

File: packages/mdc-textfield/README.md [hunk 4]
```diff
@@ -329,102 +329,155 @@ Mixin | Description
...
+`disabled` | `boolean` | Proxies to the foundation's `isDisabled`/`setDisabled` methods.  @param Sets the Text Field disabled or enabled.
+`helperTextContent` | `string` | Sets the helper text element content. Proxies to the foundation's `setHelperTextContent` method when set.
+`leadingIconAriaLabel` | `string` | Sets the aria label of the leading icon. Proxies to the foundation's `setLeadingIconAriaLabel` method.
+`leadingIconContent` | `string` | Sets the text content of the leading icon. Proxies to the foundation's `setLeadingIconContent` method.
+`max` | `string` | @param Sets the input element's max.
+`maxLength` | `number` | @param Sets the input element's maxLength.
+`min` | `string` | @param Sets the input element's min.
+`minLength` | `number` | @param Sets the input element's minLength.
+`required` | `boolean` | @param Sets the Text Field to required.
+`ripple` | `MDCRipple | null` | The `MDCRipple` instance for the root element that `MDCTextField` initializes; this only applies to the default Text Field, and is `null` for other variants.
```
Comment (moog16) [inline packages/mdc-textfield/README.md]: I need to escape the `MDCRipple | null`...this isn't rendered the same on Github.

File: packages/mdc-textfield/README.md [hunk 5]
```diff
@@ -329,102 +329,155 @@ Mixin | Description
...
+`listen(evtType: K, handler: SpecificEventListener<K>, options?: AddEventListenerOptions \| boolean) => void` | Wrapper method to add an event listener to the component's root element. This is most useful when listening for custom events.
+`unlisten(evtType: K, handler: SpecificEventListener<K>, options?: AddEventListenerOptions \| boolean) => void` | Wrapper method to remove an event listener to the component's root element. This is most useful when unlistening for custom events.
 
-## Usage Within Frameworks
+#### Properties
 
-If you are using a JavaScript framework, such as React or Angular, you can create a Text Field for your framework. Depending on your needs, you can use the _Simple Approach: Wrapping MDC Web Vanilla Components_, or the _Advanced Approach: Using Foundations and Adapters_. Please follow the instructions [here](../../docs/integrating-into-frameworks.md).
+Name | Type | Description
+--- | --- | ---
+`pattern` | `string` | @param Sets the input element's validation pattern.
```
Comment (abhiomkar) [inline packages/mdc-textfield/README.md]: Remove `@param`.

**Pull request closed without merging**